applications/suit/flash_companion/README.rst

```diff
@@ -0,0 +1,84 @@
+.. _suit_flash_companion:
+
+SUIT flash companion
+####################
+
+.. contents::
+   :local:
+   :depth: 2
+
+The flash companion application allows the secure domain firmware to access the external memory during the SUIT firmware upgrade.
+
+Application overview
+********************
+
+The flash companion implements a device driver for the external flash device and provides an IPC service to the secure domain.
+The secure domain firmware can request from the flash companion to read, erase, or write data to the external memory.
+
+The application is meant to be booted by the secure domain while performing the firmware update process using the SUIT firmware upgrade.
+
+The flash companion firmware is not a stand-alone application.
+The application must be built as a child image of a different sample as it inherits the device tree configuration from the parent build.
+
+The application can be build as a child image of your application by enabling the :kconfig:option:`SB_CONFIG_SUIT_BUILD_FLASH_COMPANION` Kconfig option.
+The devicetree configuration of the parent build will be automatically applied to the flash companion application.
```

Comment:
This should only be mentioned in the Configuration section, not in the Overview.